### ISSUE TITLE
AnnotationBbox returns weird window_extent?

### ISSUE THREAD

**Issue by znstrider** [matplotlib/matplotlib#19530]
### Bug report

**Bug summary**

A TextArea that is placed onto the figure with an AnnotationBbox at the same location as an ax.text returns a different window extent and I can't figure out why that is / how to get the window extent in the same coordinates.

Same for AnnotationBbox with xycoords=fig.transFigure and fig.text.

**Code for reproduction**

```python
import matplotlib.pyplot as plt
import matplotlib as mpl
from matplotlib.offsetbox import AnnotationBbox, TextArea

fig, ax = plt.subplots(1, 2, figsize=(6, 4))

figtext = fig.text(s='text', x=0.5, y=0.5, ha='center', va='center', color='C1')

textarea = TextArea(s='text', textprops={'color': 'None'})
annobox = AnnotationBbox(textarea, xy=(0.5, 0.5),
                         bboxprops={'facecolor': 'lightgrey', 'alpha': 0.5, 'edgecolor': 'none'})
ax[0].add_artist(annobox)

axtext = ax[0].text(s='text', x=0.5, y=0.5, ha='center', va='center', color='C1', zorder=9)

textarea2 = TextArea(s='text', textprops={'color': 'None'})
annobox2 = AnnotationBbox(textarea2, xy=(0.5, 0.5), xycoords=fig.transFigure,
                         bboxprops={'facecolor': 'lightgrey', 'alpha': 0.5, 'edgecolor': 'none'})
ax[0].add_artist(annobox2)

plt.show()

renderer = fig.canvas.get_renderer()
print(axtext.get_window_extent(renderer),'\n',
      annobox.get_window_extent(renderer))

print(figtext.get_window_extent(renderer),'\n',
      annobox2.get_window_extent(renderer))
```

**Actual outcome**

```
Bbox(x0=120.27840909090907, y0=139.72, x1=139.90340909090907, y1=149.72) 
 Bbox(x0=184.5193181818182, y0=247.34000000000003, x1=383.9625, y1=282.34000000000003)

Bbox(x0=206.1875, y0=139.0, x1=225.8125, y1=149.0) 
 Bbox(x0=356.3375, y0=245.89999999999998, x1=411.9625, y1=280.9)
```

<img width="380" alt="AnnotationBBox_example" src="https://user-images.githubusercontent.com/34557509/108106404-2f875f80-708e-11eb-84a6-e0965f7ba507.png">

**Expected outcome**

I would expect to get the same window_extents returned both times as they show above one another on the plot.

**Matplotlib version**

  * Operating system: MacOS
  * Matplotlib version 3.3.2
  * Matplotlib backend module://ipykernel.pylab.backend_inline
  * Python version: 
  * Jupyter version (if applicable): 
  * Other libraries:

**Comment by jklymak** [comment]
Your dpi must be changing at some point for some reason.  Perhaps is a problem with the inline backend.  Just doing this from a default backend works fine for me:

```
python testBBox.py
Bbox(x0=334.0511363636363, y0=382.0, x1=388.6761363636363, y1=410.0)
 Bbox(x0=322.94002525252523, y0=370.8888888888889, x1=399.78724747474746, y1=421.11111111111114)
Bbox(x0=572.6875, y0=386.0, x1=627.3125, y1=414.0)
 Bbox(x0=561.5763888888889, y0=374.8888888888889, x1=638.4236111111111, y1=425.11111111111114)
```

**Comment by znstrider** [comment]
Running a script I get roughly the same output as you.

**Comment by jklymak** [comment]
I'm not really clear why ipython would lead to a funny state in the two `get_window_extent` calls, but clearly the transform stack is different between the two and somehow not getting updated properly.

**Comment by znstrider** [comment]
It seems the canvas needs to be drawn first for the AnnotationBbox to be placed?

If I change the script to call plt.show() only after printing the window_extents I get these outputs running a script:

```
Bbox(x0=166.93181818181816, y0=191.0, x1=194.43181818181816, y1=205.0) 
 Bbox(x0=-4.0, y0=-4.0, x1=27.5, y1=11.0)
Bbox(x0=286.25, y0=193.0, x1=313.75, y1=207.0) 
 Bbox(x0=-4.0, y0=-4.0, x1=27.5, y1=11.0)
```

I can circumvent this by first calling
```
plt.show(block=False)
fig.canvas.draw()
```
and after more plotting call
```
fig.canvas.draw()
plt.show()
```
to show the expected output.

```
Bbox(x0=347.6136363636363, y0=389.0, x1=375.1136363636363, y1=403.0) 
 Bbox(x0=322.81502525252523, y0=370.8888888888889, x1=399.91224747474746, y1=421.11111111111114)
Bbox(x0=586.25, y0=393.0, x1=613.75, y1=407.0) 
 Bbox(x0=572.5625, y0=386.0, x1=627.4375, y1=414.0)
```

With the inline backend, calling fig.canvas.draw() after the AnnotationBbox was added to the axes aligns the printed output.

**Comment by rcomer** [comment]
I think this was fixed by #26184.  In a Jupyter Notebook with v3.6.2 (just released) I get these extents before and after `plt.show()`:

```
Bbox(x0=166.93181818181816, y0=191.0, x1=194.43181818181816, y1=205.0) 
 Bbox(x0=161.37626262626262, y0=185.44444444444446, x1=199.98737373737373, y1=210.55555555555557)
Bbox(x0=286.25, y0=193.0, x1=313.75, y1=207.0) 
 Bbox(x0=280.69444444444446, y0=187.44444444444446, x1=319.30555555555554, y1=212.55555555555557)
```

Please comment or re-open if you think there is an outstanding issue here.
